pkg/sql/opt/exec/execbuilder/mutation.go

```diff
@@ -331,8 +331,15 @@ func (b *Builder) buildDeleteRange(del *memo.DeleteExpr) (execPlan, error) {
 	scan := del.Input.(*memo.ScanExpr)
 	tab := b.mem.Metadata().Table(scan.Table)
 	needed, _ := b.getColumns(scan.Cols, scan.Table)
+	var indexCols opt.ColSet
+	numCols := scan.Constraint.Columns.Count()
+	for i := 0; i < numCols; i++ {
+		indexCols.Add(scan.Constraint.Columns.Get(i).ID())
+	}
+	maxKeys := int(scan.Constraint.CalculateMaxResults(
```

Comment:
This section doesn't really work right - I'm not quite sure how to construct the proper parameters to `CalculateMaxResults` here. @RaduBerinde could you please advise?